src/bun.js/bindings/KeyObject.cpp

```diff
@@ -0,0 +1,2352 @@
+            return JSC::JSValue::encode(JSCryptoKey::create(structure, globalObject, WTFMove(impl)));
+        }
+        case ArrayBufferType: {
+            auto* jsBuffer = jsCast<JSC::JSArrayBuffer*>(bufferArg.asCell());
+            if (UNLIKELY(!jsBuffer)) {
+                break;
+            }
+            auto* buffer = jsBuffer->impl();
+            void* data = buffer->data();
+            size_t byteLength = buffer->byteLength();
+            if (UNLIKELY(!data)) {
+                break;
+            }
+            Zig::GlobalObject* globalObject = reinterpret_cast<Zig::GlobalObject*>(lexicalGlobalObject);
+            auto& vm = globalObject->vm();
+            auto* structure = globalObject->JSCryptoKeyStructure();
+            auto impl = CryptoKeyHMAC::generateFromBytes(data, byteLength, CryptoAlgorithmIdentifier::HMAC, true, CryptoKeyUsageSign | CryptoKeyUsageVerify).releaseNonNull();
+            return JSC::JSValue::encode(JSCryptoKey::create(structure, globalObject, WTFMove(impl)));
+        }
+        default:
+            break;
+        }
+    }
+    {
+        auto& vm = lexicalGlobalObject->vm();
+        auto scope = DECLARE_THROW_SCOPE(vm);
+        throwException(lexicalGlobalObject, scope, createTypeError(lexicalGlobalObject, "ERR_INVALID_ARG_TYPE: expected Buffer or array-like object"_s));
+        return JSValue::encode(JSC::jsUndefined());
+    }
+}
+
+JSC::EncodedJSValue KeyObject__Exports(JSC::JSGlobalObject* globalObject, JSC::CallFrame* callFrame)
+{
+
+    auto count = callFrame->argumentCount();
+    auto& vm = globalObject->vm();
+
+    if (count < 2) {
+        auto scope = DECLARE_THROW_SCOPE(vm);
+        JSC::throwTypeError(globalObject, scope, "exports requires 2 arguments"_s);
+        return JSC::JSValue::encode(JSC::JSValue {});
+    }
+
+    if (auto* key = jsDynamicCast<JSCryptoKey*>(callFrame->argument(0))) {
+
+        auto& wrapped = key->wrapped();
+        auto key_type = wrapped.type();
+        auto id = wrapped.keyClass();
+        if (auto* options = jsDynamicCast<JSC::JSObject*>(callFrame->argument(1))) {
+            JSValue formatJSValue = options->getDirect(vm, PropertyName(Identifier::fromString(vm, "format"_s)));
+            JSValue typeJSValue = options->getDirect(vm, PropertyName(Identifier::fromString(vm, "type"_s)));
+            JSValue passphraseJSValue = options->getDirect(vm, PropertyName(Identifier::fromString(vm, "passphrase"_s)));
+            auto hasPassphrase = !passphraseJSValue.isUndefinedOrNull() && !passphraseJSValue.isEmpty();
+            if (formatJSValue.isUndefinedOrNull() || formatJSValue.isEmpty()) {
+                auto scope = DECLARE_THROW_SCOPE(vm);
+                JSC::throwTypeError(globalObject, scope, "format is expected to be a string"_s);
+                return JSC::JSValue::encode(JSC::JSValue {});
+            }
+            auto string = formatJSValue.toWTFString(globalObject);
```

Comment:
If we need to throw if it's not a string, we can do that before by calling `.isString`